sky/serve/core.py

```diff
@@ -160,17 +174,21 @@ def up(
         controller_exist = (
             global_user_state.get_cluster_from_name(controller_name)
             is not None)
-        requested_cloud = list(task.resources)[0].cloud
-        controller_cloud = (requested_cloud if not controller_exist and
-                            controller_resources.cloud is None else
-                            controller_resources.cloud)
+        if (not controller_exist and
+                controller_resources_in_config.cloud is None):
+            controller_clouds = requested_clouds
+        else:
+            controller_clouds = {controller_resources_in_config.cloud}
```

Comment:
Should we consider moving these logics into `controller_utils` and have it shared by serve and spot controller?